contracts/contracts/token/OUSD.sol

```diff
@@ -665,18 +665,16 @@ contract OUSD is Governable {
         );
         RebaseOptions stateFrom = rebaseState[from];
         RebaseOptions stateTo = rebaseState[to];
-        require(
-            _isNonRebasingAccount(from) &&
+        if (!_isNonRebasingAccount(from) &&
```

Comment:
🟠 The code changes here have deleted a "require" statement. 

The two goals of this code:

1. Have the correct accounting type going into the rest of the function.
2. Only allow the correct kinds of contracts to be delegated in, and in a way that is whitelist only for futureprofness


By removing the require, and anding the two checks into a migration, this code now has the future possibilities of failing to migrate something that should be migrated (because it might need migration, but be of future state that does return true), or it might be a state that should not be allowed, but will skip the if statement because the accounting is incorrect.

For example, if we added a double yield account type, we would want this code to reject that until we edited the code to explicitly allow for it and handle the conversion differences.

What we should do here is split the state requires on away from the conditional accounting migration.

So `require(whitelised states)` then `if(needs migration){ migrate()}`

Reply:
I didn't do the additional require, because there is already a require above checking that the accounts are not part of yield delegation. Nevertheless, I agree adding this check is a good defensive programming measure and future-proofing approach. Done here: [90ec917b0c69833de0144685856114b3759b38b6](https://github.com/OriginProtocol/origin-dollar/pull/2298/commits/90ec917b0c69833de0144685856114b3759b38b6)